File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include<math.h>

//File returns the amount of coins that are used to give back some change
//File works by geting the amount of change, converting it to cents, then chiping away at it
//with 25,10,5,1 cent coins
int main(void)
{
    
    float change = 0;
    int num_coins = 0;
    
    do 
    {
        printf("How much change is owed?  ");
        change = GetFloat();
        change *= 100;
        change = round(change);

    }
    while (change < 0);

    while (change >= 25)
    {
        change -= 25;
        num_coins ++;
    }
    
    while (change >= 10)
    {
        change -= 10;
        num_coins ++;
    }

    while (change >= 05)
    {
        change -= 05;
        num_coins ++;
    }
    
    while (change > 0)
    {
        change -= 1;
        num_coins ++;
    }

    printf("%d\n", num_coins);
    
}

File: pset6/greedy.py
import cs50

def main():
    change = 0
    num_coins = 0
    
    while True:
        print("How much change is owed? ", end="")
        change = cs50.get_float()
        change = change * 100
        change = round(change)
        if change > 0:
            break
    while True:
        if change < 25:
            break        
        change -= 25
        num_coins += 1
    while True:
        if change <= 10:
            break        
        change -= 10
        num_coins += 1
    while True:
        if change <= 5:
            break
        change -= 5
        num_coins += 1
    while True:
        if change <= 0:
            break
        change -= 1
        num_coins += 1
    print(num_coins)
if __name__ == "__main__":
    main()